tools/mtypes/main.go

```diff
@@ -0,0 +1,217 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+)
+
+type stats struct {
+	families, series, buckets, objectives int
+
+	// adjustedSeries represents series that would result in "series" in Prometheus data model
+	// (includes _bucket, _count, _sum, _quantile).
+	adjustedSeries int
+}
+
+var metricType_NATIVE_HISTOGRAM dto.MetricType = 999
+
+func main() {
+	resource := flag.String("resource", "", "Path or URL to the resource (file, <url>/metrics) containing Prometheus metric format.")
+	avalancheFlagsForTotal := flag.Int("avalanche-flags-for-adjusted-series", 0, "If more than zero, it additionally prints flags for the avalanche 0.6.0 command line to generate metrics for the similar type distribution; to get the total number of adjusted series to the given value.")
+	flag.Parse()
+
+	var input io.Reader = os.Stdin
+	if *resource != "" {
+		switch {
+		case strings.HasPrefix(*resource, "https://"), strings.HasPrefix(*resource, "http://"):
+			if _, err := url.Parse(*resource); err != nil {
+				log.Fatalf("error parsing HTTP URL to the resource %v; got %v", *resource, err)
+			}
+			resp, err := http.Get(*resource)
+			if err != nil {
+				log.Fatalf("http get against %v failed", err)
+			}
+			defer resp.Body.Close()
+			input = resp.Body
+		default:
+			// Open the input file.
+			file, err := os.Open(*resource)
+			if err != nil {
+				log.Fatalf("Error opening file: %v", err)
+			}
+			defer file.Close()
+			input = file
+		}
+	}
+	statistics, err := calculateTargetStatistics(input)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var total stats
+	for _, s := range statistics {
+		total.families += s.families
+		total.series += s.series
+		total.adjustedSeries += s.adjustedSeries
+	}
+
+	writeStatistics(os.Stdout, total, statistics)
+
+	if *avalancheFlagsForTotal > 0 {
+		adjustedGoal := float64(*avalancheFlagsForTotal)
+		fmt.Println()
+		fmt.Println("Avalanche flags for the similar distribution to get to the adjusted series goal of:", adjustedGoal)
+
+		adjustedGoal = adjustedGoal / 10.0 // Assuming --series-count=10
+		adjustedSum := 0
+		for _, mtype := range allTypes {
+			s := statistics[mtype]
+			adjustedSeriesRatio := float64(s.adjustedSeries) / float64(total.adjustedSeries)
+
+			adjustedSeries := int(adjustedGoal * adjustedSeriesRatio)
+
+			switch mtype {
+			case dto.MetricType_GAUGE:
+				fmt.Printf("--gauge-metric-count=%v\n", adjustedSeries)
+				adjustedSum += adjustedSeries
+			case dto.MetricType_COUNTER:
+				fmt.Printf("--counter-metric-count=%v\n", adjustedSeries)
+				adjustedSum += adjustedSeries
+			case dto.MetricType_HISTOGRAM:
+				avgBkts := s.buckets / s.series
+				adjustedSeries /= 2 + avgBkts
```

Comment:
Wouldn't this be:


```
adjustedSeries /= (2 + avgBkts + 1)
```

To account for the additional +inf bucket?

Reply:
Or is this more right:

```
avgBkts := ((s.buckets / s.series) - s.series) // get rid of the +inf bucket from each series
adjustedSeries /= 2 + avgBkts
```

The +inf bucket keeps throwing me off

Reply:
When I tried it for a spin it worked well so I trust the math, and its not important to be super accurate here anyway so SGTM